### DISCUSSION TITLE
Server side templates: how to implement partial updates?

### DISCUSSION THREAD

**Discussion by pwtail** [bigskysoftware/htmx#1140] [Q&A] [answered]
Hi! My question is about regular django app with templates. Suppose we are using templates mostly, no API.

Suppose we have a view that corresponds to a template that has 5 components. 2 components have figured out their data has been invalidated and they need an update. What do they do? Access their corresponding view? In that case, the view will render all 5 components, where we need only two of them.

How do you solve that problem?

**Comment by David-Guillot** [comment, marked as answer]
Hi @pwtail ,

There are multiple solutions for that:
- My favorite is to use one of the htmx [request headers](https://htmx.org/reference/#request_headers), like `Hx-Trigger` or `Hx-Target`, that are automatically set by htmx when sending a request according to your `hx-trigger` and `hx-target` attributes, and that can be parsed by your Django view (even more easily with [django-htmx](https://github.com/adamchainz/django-htmx/)) in order to decide which template to render
- Another one is to render the whole page every time and select the relevant fragment on client-side with [hx-select](https://htmx.org/attributes/hx-select). This sounds suboptimal but sometimes it helps, especially if splitting your Django templates requires a big amount of work
- Yet another approach is to use distinct URLs for each component ; this might be a little more complicated to setup on HTML side but it can help splitting your Django views...

Hope this helps,

**Reply by pwtail** [reply]
Thank you, I think, hx-select is closer to what I need
